Add com.github.hluk.copyq

hluk opened the pull request.

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+    "command": "copyq",
+
+    "tag": "v3.2.0",
+    "commit": "ac9f67ee99caa072e85951ba9b6b6f89c1e0ee39",
```
`tag` and `commit` don't go here.

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/hluk/CopyQ.git"
```
`tag` and `commit` go here =)

hluk replied:
Oh, that's why it didn't seem to do anything. :) I changed that from nightly builds on the last moment, strangely it looked like it builds the correct version anyways.

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+                    "url": "https://github.com/hluk/CopyQ.git"
+                }
+            ],
+            "config-opts": [
```
`-DCMAKE_BUILD_TYPE=Release` probably..

TingPing commented:
`https://copyq.org/` doesn't seem to be a domain you actually control? If you don't have one something like `com.github.hluk.copyq` is common.

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+            ],
+            "config-opts": [
+                "-DDESKTOP_INSTALL_NAME=org.copyq.copyq.desktop",
+                "-DAPPDATA_INSTALL_NAME=org.copyq.copyq.appdata.xml",
```
You can't just rename the file like that. The `app-id` in the file is still `copyq.desktop` which is incorrect. Also this should not be build time configurable, your application should have one id that is always used everywhere.

TingPing commented:
Your appdata file should have `<releases>` tags so software stores know its version. See also: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+    "finish-args": [
+        "--socket=x11",
+        "--socket=wayland",
+        "--env=XDG_CURRENT_DESKTOP=''"
```
`"--talk-name=org.kde.StatusNotifierWatcher"` should fix your tray icon. (The fact it depends upon this env var at all is broken, but that seems to be something to blame on a library you use)

TingPing commented on org.copyq.copyq.json:
```diff
@@ -0,0 +1,34 @@
+    "commit": "ac9f67ee99caa072e85951ba9b6b6f89c1e0ee39",
+
+    "finish-args": [
+        "--socket=x11",
```
You'll want `--share=ipc` and `--device=dri` also.

hluk commented:
Added two patches to fix the AppData file (they're already applied in the upstream git repo, though I didn't find a way to use patch URL for flatpak).

TingPing commented on fix-desktop-file-path-in-appdata.patch:
```diff
@@ -0,0 +1,47 @@
+ <!-- Copyright 2018 Lukáš Holeček <hluk@email.cz> -->
+ <component type="desktop">
+-    <id>copyq.desktop</id>
++    <id>${DESKTOP_INSTALL_NAME}</id>
```
I still strongly disagree with this approach. Your `id` is a single identifier for the project it should not be dependent on a build flag.

hluk replied:
Do I understand it correctly that the desktop file should be always `com.github.hluk.copyq.desktop`? Looking at `/usr/share/applications` on my system these files have usually simpler names.

TingPing replied:
Flatpak requires a unique ID in that format yes. But the desktop file is a different, though related, issue. The appstream file should always be in the rdns style.

TingPing commented:
> Use rename-* options instead of build flags

I'm not sure you understand my point. Your application should have a single `id`, it is used to refer to it in software managers like Plasma Discover, GNOME Software, and is used to correctly detect duplicates, to store online reviews, etc.

You pick a single id that represents your application. These `rename-` options in `flatpak-builder` are just bad hacks that exist because upstreams are not always involved in this process but since *you are upstream* you should pick an `id`.

hluk commented:
> I'm not sure you understand my point. Your application should have a single id

I understand, that makes sense. What I don't understand is why most of installed appdata and desktop files on Arch Linux have simpler name, e.g. `/usr/share/applications/hexchat.desktop`.

(I usually copy desktop files manually if I want app to autostart, but not having the file name match application name would be bit annoying.)

TingPing commented:
> I understand, that makes sense. What I don't understand is why most of installed appdata and desktop files on Arch Linux have simpler name, e.g. /usr/share/applications/hexchat.desktop.

https://github.com/flathub/org.filezillaproject.Filezilla/pull/1#issuecomment-354509330

hluk commented:
@TingPing Thanks for the clarification.

I've removed the `rename-*` stuff and added one patch. I'll merge the patch to upstream repo if it looks OK now.

nedrichards commented:
Repository has been created: https://github.com/flathub/com.github.hluk.copyq
